common_crawler/main.py

```diff
@@ -101,6 +105,75 @@ def main():
         print(f"Error while saving cache manager: {e}")
 
 
+def strip_url(url: str) -> str:
+    """Strips http(s)://www. from the beginning of a url if applicable. 
+
+    Args:
+        url (str): The URL to strip.
+
+    Returns:
+        str: The stripped URL.
+    """    
+    result = re.search(r"^(?:https?://)?(?:www\.)?(.*)$", url).group(1)
+    return result
+
+
+def remove_local_duplicates(url_results: list[str]) -> list[str]:
+    """Removes duplicate URLs from a list, ignoring http(s)://www.
+
+    Args:
+        url_results (list[str]): List of URLs to deduplicate.
+
+    Returns:
+        list[str]: List of unique URLs.
+    """    
+    stripped_url_results = [strip_url(url) for url in url_results]
+    unique_urls = collections.deque()
+    adjust = 0
+
+    for index, url in enumerate(stripped_url_results):
+        if url in unique_urls:
+            del url_results[index - adjust]
+            adjust += 1
+        else:
+            unique_urls.appendleft(url)
+
+    return url_results
+
+
+def remove_remote_duplicates(url_results: list[str]) -> list[str]:
+    """Removes URLs from a list that are already present in the Label Studio project, ignoring http(s)://www.
+
+    Args:
+        url_results (list[str]): List of URLs to deduplicate.
+
+    Returns:
+        list[str]: List of remaining URLs not present in the Label Studio project.
+    """    
+    # Retrieve URLs from remote Label Studio Project
+    config = LabelStudioConfig()
+    api_manager = LabelStudioAPIManager(config)
+    response = api_manager.export_tasks_from_project(all_tasks=True)
+    remote_results = response.json()
+    
+    try:
+        remote_urls = [strip_url(result["data"]["url"]) for result in remote_results]
+    except TypeError:
+        print("Invalid Label Studio credentials. Database could not be checked for duplicates.")
+        return url_results
```

Comment:
Three things here:
1. From the standpoint of someone reading the code (and especially someone unfamiliar with the code base), it doesn't necessarily follow that a Type Error would indicate invalid Label Studio credentials. Additionally, it strikes me as possible that there are other reasons beyond Invalid Label Studio credentials that a TypeError could appear. A better solution, I would think, is to examine the contents of the response and see if it mentions invalid credentials _specifically_, and then make such an announcement.
2. If possible, we should check for valid credentials earlier in the script. To spend at least a minute or two downloading and processing code only to find out then that one's credentials are invalid would slow down the process and probably annoy some people. 
3. I'm not sure we would want to still proceed if we can't actually check the database for duplicates. This might be a case where we actually want the code to fail gracefully, lest we add duplicates into the database.